### DISCUSSION TITLE
About Smooth reload.

### DISCUSSION THREAD

**Discussion by Murasame233** [ntex-rs/ntex#302] [General]
Here's my scenario:
- Config file contains listen address and port.
- I want to reload server at runtime.
- I don't want to break the connection that already connect to server.

How can I do it?
I tries so many ways to do it. If you use stop runtime or abort the handle manually, the connected connection will brake too.
So I tries set timeout for a connection, and when we need reload the config, we start a new runtime, and old runtime will be stop after elapsed the `timeout` value from now.

Is there any better solution?

**Comment by Murasame233** [comment]
And there's a problem, the old runtime still can access by client when its not shutdown. I make a middleware to reject the request.

**Comment by fafhrd91** [comment]
it is not possible to reload ips and ports at runtime without reloading runtime
